app/models/puppetclass.rb

```diff
@@ -41,7 +41,10 @@ class Puppetclass < ActiveRecord::Base
   scoped_search :in => :class_params, :on => :key, :complete_value => :true, :only_explicit => true
 
   scope :not_in_any_environment, -> { includes(:environment_classes).where(:environment_classes => {:environment_id => nil}) }
-
+  scope :with_possible_environments, lambda {
+    joins('LEFT OUTER JOIN environment_classes ON puppetclasses.id = environment_classes.puppetclass_id
+           LEFT OUTER JOIN environments ON environment_classes.environment_id = environments.id')
```

Comment:
why left outer join? wont you get duplicates this way? maybe I'm missing something